src/main/java/seedu/address/model/participation/Participation.java

```diff
@@ -61,8 +85,31 @@ public boolean isSameParticipation(Participation otherParticipation) {
                                         && otherParticipation.getStudent().equals(getStudent());
     }
 
+    /**
+     * @return this (not a copy)
+     */
+    public Participation copy() {
```

Comment:
Do we also need a copy function for participation class to return a new participation object like tutorial and attendance

Reply:
Cause participation have relationship with both tutorial and person. 
Participation contains tutorial and person
person contains participation
tutorial contains participation

Creating any of this object would technically involve creating new objects that they contain. So for eg, creating participation would involve creating new tutorial, but creating a new tutorial will then create new participation, the cycle will continue and it will lead to an infinite loop. Thats why I chose to make one part not creat a new object of itself. 

Maybe there are better methods to do this, will figure it out next week